Client/Client.Qt/Widgets/AbstractDropDownMenu.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <QWidget>
+#include <QMenu>
+
+class IDropDownMenu
+{
+public:
+    virtual ~IDropDownMenu() = default;
+
+    virtual void setDropDownMenu(std::unique_ptr<QMenu> newMenu) = 0;
+    virtual void showDropDownMenu(QPoint position) = 0;
+};
+
+class AbstractDropDownMenu : public QWidget, public IDropDownMenu
+{
+    Q_OBJECT
+
+public:
+    explicit AbstractDropDownMenu(QWidget *parent = nullptr);
+    ~AbstractDropDownMenu() override = default;
+
+    void setDropDownMenu(std::unique_ptr<QMenu> newMenu) override;
```

Comment:
Try to find better name. Here, you set the elements of drop down menu if I'm right, but not the DropDownMenu object

Reply:
Maybe 'setMenu' would be better?

Reply:
To avoid a misunderstanding I and Daniil already discussed this case :)
Thanks for your comment

Reply:
void setDropDownMenu(std::unique_ptr<QMenu> newMenu) override;
changed to
void setMenu(std::unique_ptr<QMenu> newMenu) override;